Tests/MessagingInApp/Views/InAppMessageViewTest.swift

```diff
@@ -421,7 +435,48 @@ class InAppMessageViewTest: UnitTest {
         XCTAssertTrue(isInlineViewVisible(inlineView))
         onCustomActionButtonPressed(onInlineView: inlineView)
 
-        XCTAssertFalse(inlineMessageDelegateMock.onInlineCustomButtonActionCalled)
+        XCTAssertFalse(inlineMessageDelegateMock.onActionClickCalled)
+        XCTAssertTrue(eventListenerMock.messageActionTakenCalled)
+        XCTAssertEqual(eventListenerMock.messageActionTakenReceivedArguments?.message, gistInAppInlineMessage)
+        XCTAssertEqual(eventListenerMock.messageActionTakenReceivedArguments?.actionValue, "Test")
+        XCTAssertEqual(eventListenerMock.messageActionTakenReceivedArguments?.actionName, "")
+    }
+
+    @MainActor
+    func test_onInlineButtonAction_multipleInlineMessages_givenDelegateSetForAll_expectSingleCustomCallback() async {
+        messagingInApp.setEventListener(eventListenerMock)
+
+        let givenInlineMessage1 = Message.randomInline
+        let gistInAppInlineMessage1 = InAppMessage(gistMessage: givenInlineMessage1)
+
+        let givenInlineMessage2 = Message.randomInline
+        queueMock.getInlineMessagesReturnValue = [givenInlineMessage1, givenInlineMessage2]
+
+        // Create two inline in app views
+        let inlineView1 = InAppMessageView(elementId: givenInlineMessage1.elementId!)
+        inlineView1.onActionDelegate = inlineMessageDelegateMock
+        let inlineView2 = InAppMessageView(elementId: givenInlineMessage2.elementId!)
+        inlineView2.onActionDelegate = inlineMessageDelegateMock
```

Comment:
```suggestion
        let inlineOnActionDelegate1 = InAppMessageViewActionDelegateMock()
        let inlineOnActionDelegate2 = InAppMessageViewActionDelegateMock()

        // Create two inline in app views
        let inlineView1 = InAppMessageView(elementId: givenInlineMessage1.elementId!)
        inlineView1.onActionDelegate = inlineOnActionDelegate1
        let inlineView2 = InAppMessageView(elementId: givenInlineMessage2.elementId!)
        inlineView2.onActionDelegate = inlineOnActionDelegate2
```

The test function looks good so far. I think the test function could test our View better if each View had it's own delegate instance. 

Then in the assertions, we verify that only 1 of the delegates was called and not the other delegate: 

```swift
XCTAssertFalse(inlineOnActionDelegate2.mockCalled)

XCTAssertTrue(inlineOnActionDelegate1.onActionClickCalled)
...
```